docs(community): update contributing page with useful links

dnilasor opened the pull request:
I'd like to update the contributing page with some more useful content than what is on there now. I've included:

- a basic welcoming message
- more clearly explained links to helpful existing resources like the "submitting a patch" page
- new links to new pages like the Development Environments page from Gardening Days, which is more up to date than the other dev environment setup guidance

This is a bit of a band-aid given the updates coming to spinnaker.io imminently, but I think this is helpful to do now. Currently if folks look at "contributing" on our site they essentially tells them that contributions aren't worked out from a licensing perspective (stated in a detracting tone) and that if they REALLY want to contribute they should reach out on Slack. I've reframed this to put the (valid) acknowledgment that we don't know what we're doing fully re: licensing at the bottom of the page, while continuing to encourage new contributors to join Slack from the beginning of the page. 

This page will be more useful immediately for Spinnaker contributors browsing our site who want to participate in the CDF Hacktoberfest and the upcoming Summit Gardening Days.

dnilasor committed:
Merge pull request #1 from spinnaker/master

update fork

dnilasor committed:
Merge remote-tracking branch 'upstream/master'

dnilasor committed:
Merge remote-tracking branch 'upstream/master'

dnilasor committed:
Merge remote-tracking branch 'upstream/master'

dnilasor committed:
Merge remote-tracking branch 'upstream/master'

dnilasor committed:
Merge remote-tracking branch 'upstream/master'

dnilasor committed:
Merge remote-tracking branch 'upstream/master'

dnilasor committed:
Merge remote-tracking branch 'upstream/master'

dnilasor committed:
Merge remote-tracking branch 'upstream/master'

dnilasor committed:
Merge remote-tracking branch 'upstream/master'

dnilasor committed:
update contributing page with useful links

dnilasor committed:
add welcome message

aimeeu committed:
copy edits

aimeeu reviewed (approved):
LGTM; built locally to verify rendered output

dorbin commented on community/contributing/index.md:
```diff
@@ -8,22 +8,34 @@ redirect_from: /docs/contributing-to-spinnaker
+## Quick Links
 
-## How to Become a Contributor
+- [Spinnaker GitHub issue tracker](https://github.com/spinnaker/spinnaker/issues): file Spinnaker issues
```
I think I'm in favor of having something like "File an issue" as the heading, and "Spinnaker issue tracker" as the link text.

dnilasor replied:
good suggestion

dorbin commented on community/contributing/index.md:
```diff
@@ -8,22 +8,34 @@ redirect_from: /docs/contributing-to-spinnaker
 
-## How to Become a Contributor
+- [Spinnaker GitHub issue tracker](https://github.com/spinnaker/spinnaker/issues): file Spinnaker issues
+- [Development environments](/community/gardening/dev-environment/): set up a local development environment to test your pull requests
```
I trust that the dev environment setup is suitable for all uses, but because it's under Gardening Days, it might be a little confusing to list it here. Would it be better to move the env setup doc to somewhere outside of Gardening Days, and link to it from there and from here?

dnilasor replied:
@dorbin yes that would make sense but I really didn't want to spend a lot of time on the site organization at this point given that it's about to be blown up. I do like the idea of moving that page out of the Gardening area but it would require some edits and I wasn't sure what to do with the existing page.

dnilasor replied:
I can try to work that into this PR if desired

dorbin replied:
It does seem like a lot of work to do right before a whole re-do. Let it stand :)

dorbin commented on community/contributing/index.md:
```diff
@@ -8,22 +8,34 @@ redirect_from: /docs/contributing-to-spinnaker
+- [Development environments](/community/gardening/dev-environment/): set up a local development environment to test your pull requests
+- [How to submit a patch](/community/contributing/submitting/): read guidelines on submitting patches, proposing new features, and using commit and PR naming conventions
+- [Code conventions for server-side components](/community/contributing/back-end-code/): read code conventions for pull requests to core services
+- [Reviewing Pull Requests](/community/contributing/docs/reviewing-prs): get started contributing by reviewing documentation pull requests
```
```suggestion
- [Reviewing pull requests](/community/contributing/docs/reviewing-prs): get started contributing by reviewing documentation pull requests.
```

dorbin replied:
And because these list items are all complete sentences, go ahead and put periods after each.

dorbin commented on community/contributing/index.md:
```diff
@@ -8,22 +8,34 @@ redirect_from: /docs/contributing-to-spinnaker
+
+## License agreements
+
+We should certainly have a place where we tell you about Contributor License Agreements and how you need to identify the right one and sign it before you are able to contribute code to Spinnaker. We'll probably end up doing something like this, but at the moment we're exploring options that might make it easier for most of you to contribute. Please stay tuned...
```
Are we ready yet to say/do anything about licensing here? Should we engage the Foundation to get some language?

dnilasor replied:
@dorbin I tried engaging the foundation to get language some time ago. I could try again. It's a process. My vote would be to remove this altogether at this point - we don't have any valuable information to share, and folks are already contributing using the standard Apache license anyway. 

What do you think of removing this section for now? I or the upcoming CX SIG could engage with the foundation about this later this year, after the Summit (which is currently consuming much of my time).

dorbin replied:
SGTM, leaving it out for now. Hopefully we can say something meaningful soon, but in the meantime, ok to delete, as far as I'm concerned.

dorbin commented on community/contributing/index.md:
```diff
@@ -8,22 +8,34 @@ redirect_from: /docs/contributing-to-spinnaker
+
+We should certainly have a place where we tell you about Contributor License Agreements and how you need to identify the right one and sign it before you are able to contribute code to Spinnaker. We'll probably end up doing something like this, but at the moment we're exploring options that might make it easier for most of you to contribute. Please stay tuned...
+
+In the meantime, you can currently contribute according to the Apache license at your own risk.
```
What are the risks? I think the language we had before was a little less...alarming?

dnilasor replied:
@dorbin good question - I do not know the risks, which was my motivation for writing this. The language before basically is a tacit encouragement for folks to NOT contribute and that's what I want to avoid:

```
This is normally the place where we’d tell you about Contributor License Agreements and how you need to identify the right one and sign it before you are able to contribute code to Spinnaker.

We’ll probably end up doing something like this, but at the moment we’re exploring options that might make it easier for most of you to contribute. Please stay tuned…

In the meantime, if you’re really excited about contributing and don’t want to wait until we have all of this worked out, please reach out to us on Slack. We love accepting community patches!
```

I don't like that this leaves it as....we don't have this figured out so you should wait for us to do that or else bug us on Slack. Doesn't seem scalable or particularly professional. But I agree that my wording is alarming. Do you have any suggestions? I got started on a project to look into the licensing stuff so we could give people real information but it wasn't anyone else's priority and I don't have the bandwidth. Should we just not talk about licensing altogether?

dorbin replied:
@plumpy, what say ye?

dorbin reviewed (commented):
Hi, @dnilasor, thanks for doing this! 
I added a few comments.

dnilasor committed:
Dorbin - update list format and add periods

dnilasor committed:
Dorbin - capitalization convention

dnilasor committed:
remove unhelpful licensing agreements text

We plan to re-add a similar feature once we have valid information about licensing in our project from the Linux Foundation (copying Jenkins is likely a great place to start)

317brian committed:
Merge branch 'master' into dnilasor-contributing-update
